spring-web/src/main/java/org/springframework/http/HttpHeaders.java

```diff
@@ -1387,6 +1425,45 @@ static String encodeHeaderFieldParam(String input, Charset charset) {
 		return sb.toString();
 	}
 
+	/**
+	 * Decode the given header field param as describe in RFC 5987.
+	 * <p>Only the US-ASCII, UTF-8 and ISO-8859-1 charsets are supported.
+	 * @param input the header field param
+	 * @return the encoded header field param
+	 * @see <a href="https://tools.ietf.org/html/rfc5987">RFC 5987</a>
+	 */
+	static String decodeHeaderFieldParam(String input) {
```

Comment:
Also I'm wondering if it's worth having a ContentDisposition representation along the lines of `HttpHeaders#getContentDisposition()`, such as [for example](https://jersey.java.net/apidocs/2.8/jersey/org/glassfish/jersey/media/multipart/ContentDisposition.html).